packages/types/test/unit/ExecutionAddress.test.ts

```diff
@@ -0,0 +1,61 @@
+import {describe, it, expect} from "vitest";
+import {toChecksumAddress} from "../../src/utils/ExecutionAddress.js";
+
+describe("toChecksumAddress", () => {
+  it("should fail with invalid addresses", () => {
+    expect(() => toChecksumAddress("0x1234")).toThrowError("Invalid address");
+  });
+
+  it("should format address as ERC55", () => {
+    type TestCase = {
+      address: string;
+      checksumedAddress: string;
+    };
+
+    expect(toChecksumAddress("52908400098527886E0F7030069857D2E4169EE7")).toBe(
+      "0x52908400098527886E0F7030069857D2E4169EE7"
+    );
+
+    const testCases: TestCase[] = [
+      // All caps
+      {
+        address: "0x52908400098527886E0F7030069857D2E4169EE7",
+        checksumedAddress: "0x52908400098527886E0F7030069857D2E4169EE7",
+      },
```

Comment:
This test case doesn't make much sense, isn't the address and the checksum address the same?

What the ERC test cases mean by "All caps" is that the checksum address has only capital letters.

I would suggest to have `address` always all lowercase as this it was will be return from `super.toJson(value)`, and we anyways convert it to lowercase right away